File: backend/app/schemas/allocation.py
from pydantic import BaseModel


class SleeveAllocationItem(BaseModel):
    sleeve: str
    market_value_base: float
    weight_of_total_nav: float
    position_count: int
    unrealized_pnl_base: float


class CountryAllocationItem(BaseModel):
    country: str
    market_value_base: float
    weight_of_total_nav: float


class SectorAllocationItem(BaseModel):
    sector: str
    market_value_base: float
    weight_of_total_nav: float


class CurrencyAllocationItem(BaseModel):
    currency: str
    total_base_value: float
    weight_of_total_nav: float

File: frontend/src/types/allocation.ts
export type SleeveAllocationItem = {
  sleeve: string;
  equity_value_base: number;
  cash_value_base: number;
  total_base_value: number;
  weight_of_total_nav: number;
  position_count: number;
  unrealized_pnl_base: number;
};

export type CountryAllocationItem = {
  country: string;
  market_value_base: number;
  weight_of_total_nav: number;
};

export type SectorAllocationItem = {
  sector: string;
  market_value_base: number;
  weight_of_total_nav: number;
};

export type CurrencyAllocationItem = {
  currency: string;
  total_base_value: number;
  weight_of_total_nav: number;
};